lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -20,21 +22,46 @@ class Scripts extends Gatherer {
   async afterPass(passContext, loadData) {
     const driver = passContext.driver;
 
-    /** @type {Object<string, string>} */
-    const scriptContentMap = {};
+    /** @type {LH.Artifacts['Scripts']} */
+    const scripts = [];
+
+    /** @type {string[]} */
+    const inlineScripts = await driver.evaluateAsync(`(() => {
+      ${getElementsInDocumentString};
+
+      return getElementsInDocument('script')
+        .filter(meta => !meta.src && meta.text.trim())
+        .map(meta => meta.text);
+    })()`, {useIsolation: true});
+
+    if (inlineScripts.length) {
+      const mainResource = loadData.networkRecords.find(
+        request => URL.equalWithExcludedFragments(request.url, passContext.url));
+      if (!mainResource) {
+        throw new Error('could not locate mainResource');
```

Comment:
I'm a teensy bit worried that this is going to start throwing errors into our perf category on edge cases we previously didn't really care about. We don't actually *need* a request ID for the whole thing to work. WDYT about making `requestId` optional?

Reply:
If I knew them I would fix them :)

I just meant that we have seen this error popup in peoples reports before and it's mostly 🤷‍♂️and move on since it wasn't that big a deal. Now it'll be a bit more user-facing. 

I actually had to tackle this for canonical though and think it'll be solid.

https://github.com/GoogleChrome/lighthouse/pull/7080/files#diff-2113a8215d43931339dbb50287d89dcbR450

Reply:
done. thoughts on handling reporting of unminified scripts in the JS audit? I just put `?` as a placeholder url :)

Reply:
also, one of us should extract that function to somewhere else (whoever merges last)

Reply:
extract it off of `network-analyzer`? that actually seems like a good place for it :)

though if you mean we should finally move `network-analyzer` out of the `depdency-graph/simulator` folder then I agree with you :)

Reply:
> done. thoughts on handling reporting of unminified scripts in the JS audit? I just put ? as a placeholder url 

Ideally we'd try to provide a snippet of the code like we do for CSS, but I'm OK with this for now until we see how common it is.

Reply:
maybe when the fancy code snippet preview (#6901) lands we can do something cool here :D

Reply:
yeah having it in `network-analyzer` is perfect

💯 to snippets